app/shared/design-system/components/primitives/auto-input.tsx

```diff
@@ -0,0 +1,205 @@
+export class AutoInput extends EventTarget {
+	// PRIVATE
+	#selectTarget = null;
+	#activeTarget = null;
+	#activeEvents = ['keydown', 'focusin', 'focusout', 'pointerover', 'pointerout', 'pointerup', 'pointerdown'];
+	#passiveEvents = ['pointermove'];
+	#resizeCursor = 'ns-resize';
+	#pointerCursor = 'pointer';
+	#pointerCapture = false;
+	#ownerDocument = globalThis.document;
+	#canvasContext = globalThis.document?.createElement('canvas').getContext('2d');
+	#typeOfValue(value) {
+		if (/^([-+]?\d*\.?\d+)$/.test(value)) return 'number';
+		if (/^(%|cm|mm|in|px|pt|pc|em|ex|ch|rem|vw|vh|vmin|vmax|vb|vi|svw|svh|lvw|lvh|dvw|dvh|lh|rlh)$/.test(value)) return 'unit';
+		if (/^(\w+)$/.test(value)) return 'keyword';
```

Comment:
this is probably not gonna work for everything, but may be good enough for now, stuff like CSS functions e.g. var()

Reply:
would a add a todo at least for now